net/libfab/libfab.c

```diff
@@ -997,16 +1002,20 @@ static void libfab_poller(struct m0_fab__tm *tm)
 				 * handled by the loop.
 				 */
 				err = ret < 0 ? -errno : 0;
-				M0_ASSERT_INFO(M0_IN(ret, (-1, 0, 1)),
-					       "Unexpected rc epoll_wait: %d",
+				if (!M0_IN(ret, (-1, 0, 1)))
+					M0_LOG(M0_ERROR,
+					       "Unexpected epoll_wait rc=%d",
 					       ret);
-				M0_ASSERT_INFO(ergo(ret < 0, err == -EINTR),
-					       "Unexpected epoll_wait err: %d",
+				if (ret == -1 && err != -EINTR)
```

Comment:
Earlier condition was ret < 0 and now its ret == -1, this is based on debug data or making condition more strict to avoid too much error logs?

Reply:
As per man page of epoll_wait(), the return value of epoll_wait() is -1 in case of error.
Hence checking `ret == -1`